modules/system/classes/PluginBase.php

```diff
@@ -432,7 +432,13 @@ public function getPluginVersion(): string
 
         $versionFile = $this->getPluginPath() . '/updates/version.yaml';
 
-        if (!File::isFile($versionFile) || !($versionInfo = Yaml::parse(file_get_contents($versionFile))) || !is_array($versionInfo)) {
+        if (
+            !File::isFile($versionFile) ||
+            !($versionInfo = Yaml::withProcessor(new VersionYamlProcessor, function ($yaml) use ($versionFile) {
+                return $yaml->parseFile($versionFile);
+            })) ||
+            !is_array($versionInfo)
+        ) {
```

Comment:
I think the PSR guidelines recommend boolean operators being at the start of the line if conditions are multi-line.

```suggestion
            !File::isFile($versionFile)
            || !($versionInfo = Yaml::withProcessor(new VersionYamlProcessor, function ($yaml) use ($versionFile) {
                return $yaml->parseFile($versionFile);
            }))
            || !is_array($versionInfo)
        ) {
```

Reply:
Got a link to that? I've always done it the above way so that's probably what's present in the core already. I don't think I've seen it done the other way in the core before.

Reply:
It's defined in PSR-12 (https://www.php-fig.org/psr/psr-12/) although we've only declared to follow PSR-2 at the moment which doesn't define it.

However, it does say they can be at the beginning or end of a line, but not both:

> Boolean operators between conditions MUST always be at the beginning or at the end of the line, not a mix of both.

Their examples show them at the start of the line, but that's probably the preference of the author.

Personally, I like them at the start because it reads to me like:

- Condition
- OR Condition
- AND Condition

I've also deployed code with this format - eg: [here](https://github.com/wintercms/winter/blob/develop/modules/backend/classes/BackendController.php#L80-L84) and [here](https://github.com/wintercms/winter/blob/develop/modules/backend/widgets/Filter.php#L164-L168). Others have [done so](https://github.com/wintercms/winter/blob/develop/modules/backend/widgets/MediaManager.php#L1414-L1417) as well.

From what I can tell, I think Sam and Alexey used your style mostly.

I don't mind either way - I think we just need to make a ruling, adjust everything to suit and then update the code checks to suit.

Reply:
Fair points, I like your reasoning for going with that approach, lets standardize on that and add it to the styling standards.